airflow/operators/python.py

```diff
@@ -145,6 +145,69 @@ def execute_callable(self):
         return self.python_callable(*self.op_args, **self.op_kwargs)
 
 
+class PythonFunctionalOperator(PythonOperator):
+    """
+    Allows a workflow to "branch" or follow a path following the execution
+    of this task.
+
+    It derives the PythonOperator and expects a Python function that returns
+    a single task_id or list of task_ids to follow. The task_id(s) returned
+    should point to a task directly downstream from {self}. All other "branches"
+    or directly downstream tasks are marked with a state of ``skipped`` so that
+    these paths can't move forward. The ``skipped`` states are propagated
+    downstream to allow for the DAG state to fill up and the DAG run's state
+    to be inferred.
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        multiple_outputs: bool = False,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.multiple_outputs = multiple_outputs
+        
+
+    def __call__(self, *args, **kwargs):
```

Comment:
probably this is the main discussion for this PR:   @task => Operator instance approach 
A. we need to make sure that the user doesn't use the same operator twice: 
```
@task
def some_task(a):
       pass

a = some_task(1)
b = some_task(2).  #second time usage with different param - Error, it will change first call as well
```

Solution:  error message on more than one __call__ and make sure user calls .alias on every extra call. 

B. if dag.task decorator is not used, the operator has to be assigned to dag.
       if user writes some_task() in dag scope, he probably expects it to be assigned to current `with DAG` context, ( if @task is defined inside the context, it will happen automatically), but usually, people will not like to define functions inside `with DAG` context.
C. right now a user is not able to change the Operator configuration around some function ( BaseOperator has 20 different params, so it's not something hypothetical: `pool`, `start_day` and others
    Solution: something similar to .alias with copy ctor that can modify specific kwargs.


I am probably too biased to suggest some specific approach here. I like current @task implementation as it keeps everything very simple, at the same time "@task as factory"  approach can be much more flexible in the future.

Reply:
Yup. This definetly needs some extra work. I want to make sure we are all aligned in the conversation around `task -> PythonFunctionalOperator` vs `task -> factory of PythonFunctionalOperator`

A. Totally agree, I can add this.
B. We can get DAG from DAGContext similarly as its done in initialization on `__call__` method if operator is not assigned and assign it.
C. True, we need to capture the *args, **kwargs in the beginning and pass it to the copy as well. (or maybe copy all class attributes? not sure what's the cleanest approach here).

Reply:
B. this is exactly what is missing.
C. probably there is no "cleanest" one, we will have to compromise. so whatever you do, it will work.